lib/Controller/SettingsController.php

```diff
@@ -78,43 +91,34 @@ public function getSamlProviderSettings($providerId) {
 				// as this is the only category that has the saml- prefix on config keys
 				if (strpos($category, 'attribute-mapping') === 0) {
 					$category = 'attribute-mapping';
-					$key = $prefix . 'saml-attribute-mapping' . '-' . $setting;
+					$key = 'saml-attribute-mapping' . '-' . $setting;
+				} else {
+					$key = $category . '-' . $setting;
+				}
+
+				if (isset ($details['global']) && $details['global']) {
+					$settings[$category][$setting] = $this->config->getAppValue('user_saml', $key, '');
```

Comment:
I'm guessing this exists only for migration purposes, maybe add a PHP doc entry to clarify this